File: test/ayush.dev/scala/pipelines/scala_external_pipeline/code/src/main/scala/io/prophecy/pipelines/scala_external_git_pipeline/graph/customers_orders.scala
package io.prophecy.pipelines.scala_external_git_pipeline.graph

import io.prophecy.libs._
import io.prophecy.pipelines.scala_external_git_pipeline.config.ConfigStore._
import org.apache.spark._
import org.apache.spark.sql._
import org.apache.spark.sql.functions._
import org.apache.spark.sql.types._
import org.apache.spark.sql.expressions._
import java.time._

object customers_orders {

  def apply(spark: SparkSession): DataFrame =
    spark.read
      .format("csv")
      .option("header",                   true)
      .option("sep",                      ",")
      .option("ignoreLeadingWhiteSpace",  false)
      .option("ignoreTrailingWhiteSpace", false)
      .schema(
        StructType(
          Array(
            StructField("customer_id",         IntegerType, true),
            StructField("orders",              LongType,    false),
            StructField("amounts",             DoubleType,  true),
            StructField("account_length_days", IntegerType, true)
          )
        )
      )
      .load("dbfs:/Prophecy/abhisheks+e2etests@prophecy.io/CustomersOrders.csv")

}

File: test/ayush.dev/python/pipelines/TestNew1/code/testnew1/graph/customers_orders.py
from pyspark.sql import *
from pyspark.sql.functions import *
from pyspark.sql.types import *
from testnew1.config.ConfigStore import *
from testnew1.udfs.UDFs import *

def customers_orders(spark: SparkSession) -> DataFrame:
    return spark.read\
        .schema(
          StructType([
            StructField("customer_id", IntegerType(), True), StructField("orders", LongType(), False), StructField("amounts", DoubleType(), True), StructField("account_length_days", IntegerType(), True)
        ])
        )\
        .option("header", True)\
        .option("sep", ",")\
        .option("ignoreLeadingWhiteSpace", True)\
        .option("ignoreTrailingWhiteSpace", True)\
        .csv("dbfs:/Prophecy/abhisheks+e2etests@prophecy.io/CustomersOrders.csv")
